### PR TITLE
docs: move flannel guide to official documentation

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6958)

### CODE REVIEW

**Comment by aanm** [comment]
test-docs-please

File: Documentation/gettingstarted/flannel-integration.rst [hunk 1]
```diff
@@ -0,0 +1,127 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    http://docs.cilium.io
+
+.. _flannel-integration:
+
+**************************************
+Cilium integration with Flannel (beta)
```
Comment (ianvernon) [inline Documentation/gettingstarted/flannel-integration.rst:10]: We have used 'tech-preview' elsewhere - what is the agreed upon terminology for determining the maturity of these sorts of features? We should be consistent cc: @tgraf

File: Documentation/gettingstarted/flannel-integration.rst [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+This guide contains the necessary steps to run Cilium on top of your Flannel
+cluster.
+
+If you have a cluster already set up with Flannel you will not need to install
+Flannel again.
+
+This Cilium integration with Flannel was performed with Flannel 0.10.0 and
+Kubernetes >= 1.9. If you find any issues with previous Flannel versions please
+feel free to reach us out to help you.
```
Comment (ianvernon) [inline Documentation/gettingstarted/flannel-integration.rst]: Grammar: reach us out --> reach out to us

File: Documentation/gettingstarted/flannel-integration.rst [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  .. group-tab:: K8s 1.8
+
+    .. parsed-literal::
+
+      curl -LO \ |SCM_WEB|\/examples/kubernetes/1.8/cilium.yaml
+
+Edit the ConfigMap in that file and set the option ``flannel-master-device`` with ``"cni0"``.
+
+Set ``flannel-uninstall-on-exit`` with either ``true`` or ``false``. If you
+plan to deploy Cilium and ensure the policy enforcement will always be kept even
```
Comment (ianvernon) [inline Documentation/gettingstarted/flannel-integration.rst]: Phrasing: "ensure that policy enforcement will persist even if you remove Cilium"

File: Documentation/gettingstarted/flannel-integration.rst [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+if you remove Cilium, then leave the option with ``false``. If you plan to test
```
Comment (ianvernon) [inline Documentation/gettingstarted/flannel-integration.rst]: Phrasing: "then leave the option set to \`\`false\`\`"

File: Documentation/gettingstarted/flannel-integration.rst [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+Cilium in your cluster and remove Cilium once you have finish your tests, by
```
Comment (ianvernon) [inline Documentation/gettingstarted/flannel-integration.rst]: Grammar: finish --> finished

Comment (ianvernon) [inline Documentation/gettingstarted/flannel-integration.rst]: Grammar: remove "by" from this sentence

File: Documentation/gettingstarted/flannel-integration.rst [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+  flannel-uninstall-on-exit: "false"
+  # Installs a BPF program to allow for policy enforcement in already running
+  # containers managed by Flannel.
+  # NOTE: This requires Cilium DaemonSet to be running in the hostPID.
+  # To run in this mode in Kubernetes change the value of the hostPID from
+  # false to true. Can be found under the path `spec.spec.hostPID`
+  flannel-manage-existing-containers: "false"
+
+
+Once you have changed the configuration map accordingly, you can deploy Cilium.
```
Comment (ianvernon) [inline Documentation/gettingstarted/flannel-integration.rst]: Nit: configuration map --> ConfigMap ?

File: Documentation/gettingstarted/flannel-integration.rst [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+.. parsed-literal::
+
+    kubectl create -f ./cilium.yaml
+
+Cilium might not come up immediately on all nodes, since flannel only sets up
```
Comment (ianvernon) [inline Documentation/gettingstarted/flannel-integration.rst]: flannel --> Flannel

File: Documentation/gettingstarted/flannel-integration.rst [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+the bridge network interface that connects containers with the outside world
+when the first container is created on that node, Cilium will wait until that
```
Comment (ianvernon) [inline Documentation/gettingstarted/flannel-integration.rst]: Run-on sentence: change to "on that node. In this case, Cilium will wait..."

**Review by ianvernon — DISMISSED** [review]

**Comment by aanm** [comment]
test-docs-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**